src/zenml/utils/source_utils.py

```diff
@@ -277,27 +280,7 @@ def resolve_class(class_: Type[Any]) -> str:
     if is_standard_source(initial_source):
         return resolve_standard_source(initial_source)
 
-    try:
-        file_path = inspect.getfile(class_)
-    except TypeError:
-        # builtin file
-        return initial_source
-
-    if (
-        initial_source.startswith("__main__")
-        or not is_inside_repository(file_path)
-        or is_third_party_module(file_path)
-    ):
-        return initial_source
-
-    # Regular user file inside the repository -> get the full module
-    # path relative to the repository
-    module_source = get_module_source_from_file_path(file_path)
-
-    # ENG-123 Sanitize for Windows OS
-    # module_source = module_source.replace("\\", ".")
-
-    return module_source + "." + class_.__name__
+    return initial_source
```

Comment:
I think this will "break" the post-execution workflow in the following way:
For each output artifact, we store the resolved string (e.g. `some.module.SomeClass`) of the materializer and datatype inside the metadata store (see `zenml/steps/utils.py` method `resolve_output_artifact`). When trying to read an artifact in post-execution, we load those classes to read the artifact. 
With the source resolved like this, not only would post-execution only work from the exact same directory as you ran the pipeline, but any materializer/custom data type defined in the main module will fail to import.

Reply:
Same as above, I restored the original code, with a code path that allows users to run zenml outside of a .zen repository. When no .zen repo is found, the "root" path against which the module source is computed is the folder where the main module file is located